java/integTest/src/test/java/glide/ConnectionTests.java

```diff
@@ -26,4 +39,184 @@ public void cluster_client() {
+        assumeTrue(SERVER_VERSION.isGreaterThanOrEqualTo("8.0.0"), "Skip for versions below 8");
+
+        String az = "us-east-1a";
+
+        // Create client for setting the configs
+        GlideClusterClient configSetClient =
+                GlideClusterClient.createClient(commonClusterClientConfig().requestTimeout(2000).build())
+                        .get();
+        assertEquals(configSetClient.configResetStat().get(), OK);
+
+        // Get Replica Count for current cluster
+        var clusterInfo =
+                configSetClient
+                        .customCommand(
+                                new String[] {"INFO", "REPLICATION"},
+                                new RequestRoutingConfiguration.SlotKeyRoute("key", PRIMARY))
+                        .get();
+        int nReplicas =
+                Integer.parseInt(
+                        Stream.of(((String) clusterInfo.getSingleValue()).split("\\R"))
+                                .map(line -> line.split(":", 2))
+                                .filter(parts -> parts.length == 2 && parts[0].trim().equals("connected_slaves"))
+                                .map(parts -> parts[1].trim())
+                                .findFirst()
+                                .get());
+        int nGetCalls = 3 * nReplicas;
+        String getCmdstat = String.format("cmdstat_get:calls=%d", 3);
+
+        // Setting AZ for all Nodes
+        configSetClient.configSet(Map.of("availability-zone", az), ALL_NODES).get();
+        configSetClient.close();
+
+        // Creating Client with AZ configuration for testing
+        GlideClusterClient azTestClient = createAzTestClient(az);
+        ClusterValue<Map<String, String>> azGetResult =
+                azTestClient.configGet(new String[] {"availability-zone"}, ALL_NODES).get();
+        Map<String, Map<String, String>> azData = azGetResult.getMultiValue();
+
+        // Check that all replicas have the availability zone set to the az
+        for (var entry : azData.entrySet()) {
+            assertEquals(az, entry.getValue().get("availability-zone"));
+        }
+
+        // set test data and perform GET operations
+        azTestClient.set("foo", "value").get();
+
+        // execute GET commands
+        for (int i = 0; i < nGetCalls; i++) {
+            azTestClient.get("foo").get();
+        }
+
+        ClusterValue<String> infoResult =
+                azTestClient
+                        .info(new InfoOptions.Section[] {InfoOptions.Section.COMMANDSTATS}, ALL_NODES)
+                        .get();
+        Map<String, String> infoData = infoResult.getMultiValue();
+
+        // Check that all replicas have the same number of GET calls
+        int count = 0;
+        for (var entry : infoData.entrySet()) {
+            System.out.println(entry.getValue());
+            if (entry.getValue().contains(getCmdstat)) {
+                count++;
```

Comment:
you can use stream.filter()count() to get this count, and it would take one line. 
reference: https://www.baeldung.com/java-stream-filter-count#2-using-count-with-filter